lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -56,6 +55,28 @@ def initialize(uri, emitter, message_queue, test_library)
           :after_call,
           :on_vcall,
         )
+
+        register_external_listeners!
+      end
+
+      sig { void }
+      def register_external_listeners!
+        self.class.listeners.each do |l|
+          @external_listeners << T.unsafe(l).new(@uri, @emitter, @message_queue)
+        end
+      end
+
+      sig { void }
+      def merge_external_listeners_responses!
+        @external_listeners.each do |l|
+          merge_response!(l)
+        end
+      end
+
+      sig { override.returns(ResponseType) }
+      def response
+        merge_external_listeners_responses!
```

Comment:
How about we invoke `merge_external_listeners_responses!` ourselves in the `Executor` and do not change `response`?

I find it a bit weird that invokeing `request.response` is no longer idempotent (since it always merges the responses).

Reply:
Make sense. Updated 👍